docs/mkdocs/documentation/day0_day1_handover.md

```diff
@@ -0,0 +1,83 @@
+
+## Definitions
+
+### Day 0
+
+The most basic kmods that are required for a node to become “Ready” in the cluster
+
+**Examples**
+* A storage driver that is required in order to mount the rootFS as part of the boot process.
+  Vendors will usually work closely with the RHEL team to make those drivers
+  in-tree so we won’t worry about them too much here.
+* A network driver that is required for the machine to access machine-config-server
+  on the bootstrap node to pull the ignition and join the cluster
+
+### Day 1
+
+Kmods that are not required for a node to become “Ready” in the cluster but are
+preferred to be installed/updated before the node is becoming “Ready”.
+
+**Examples**
+* An OOT network driver that replaces an outdated in-tree driver to exploit the full potential of the NIC.
+
+### Day2
+
+Kmods that can be dynamically loaded to the kernel or removed from it without
+interfering with the cluster infrastructure (such as connectivity).
+
+**Examples**
+* GPU operator
+* Secondary network adapters
+* FPGA
+
+## Layering background
+
+When a day0 kmod was installed in the cluster, it means that “layering” was applied
+through MCO and OCP upgrades won’t trigger node upgrades.
+
+Unless a user wants to add new features to its driver, we will never need to
+recompile it for him since the node’s OS will remain.
+
+With that being said, MCO has [plans](https://issues.redhat.com/browse/MCO-665)
+to rebuild the node images upon a cluster upgrade when Laying is used by MCO.
+
+## Using KMM for managing day0 and day1 kmods
+
+We can leverage KMM to manage the lifecycle of day0/1 kmods without a reboot when the driver allows it.
+
+### 1st option
+
+By treating the kmod as an in-tree driver.
+
+Nothing to do until user wishes to update the kmods.
+
+When user wishes to upgrade the kmod, he treats the kmod as an in-tree driver
+and creates a `Module` in the cluster with the `intreeRemoval` field to unload
+the old version of the driver.
+
+**Characteristics**
+
+* Down time - KMM will try to unload and load the kmod on all the selected nodes simultaneously.
+* Works in case removing the driver makes the node lose connectivity (because KMM uses a single pod to unload+load the driver)
+* Doesn’t work if the upgrade require a node reboot (when rebuilding initramfs is needed for example)
+
+### 2nd option
+
+By using ordered upgrade.
+
+In this case, user creates a versioned `Module` in the cluster representing the kmods - nothing
+will happen since the kmods are already loaded.
+
+When user wishes to upgrade the kmod, he uses the ordered-upgrade feature.
+
+**Characteristics**
+
+* No cluster downtime - KMM will make the upgrade node by node
+* Doesn't work if unloading the driver results in losing connection to the node
+  (because KMM will create 2 different worker pods, one for unloading and another for loading which won’t be scheduled)
+* Doesn’t work if the upgrade require a node reboot (when rebuilding initramfs is needed for example)
```

Comment:
I think it does.
- KMM unloads old driver
- User reboots node
- Node boots with the old-old driver
- KMM loads new driver

Reply:
You have a few assumption here (which doesn't always line up)
1. If it's a day0 kmod, you don't have an "old-old driver" - without the new driver - the node won't connect to the cluster
2. You assume that the kmod can be loaded with `modprobe` as a day2 operation - this is not the case with day0 and some day1 kmods.

If you need the driver in the initramFS image, there is no other option, than to re-build the initramFS image, let MCO know about the change and reboot with the new initramFS image (KMM doesn't know how to handle any of those steps)

Reply:
If it's a day0 kmod you don't have any "old-old driver" in the node's image.

Reply:
Do you need to have it?